presto-docs/src/main/sphinx/connector/hive-security.rst

```diff
@@ -382,3 +386,88 @@ node.
 
 You should ensure that the keytab files have the correct permissions on every
 node after distributing them.
+
+File Based Authorization
+========================
+
+The config file is specified in JSON and is made up of three main sections, each of
+which is a list of rules which are matched first to last. The privileges from the
+first rule that matches are used. All regexes default to ".*", if not specified.
+
+Schema Rules
+------------
+
+These rules govern who is considered an owner of a schema.
+
+* ``user`` (optional): regex to match against user name.
+
+* ``schema`` (optional): regex to match against schema name.
+
+* ``owner`` (required): boolean indicating ownership.
+
+Table Rules
+-----------
+
+These rules govern the privileges granted on specific tables.
+
+* ``user`` (optional): regex to match against user name.
+
+* ``schema`` (optional): regex to match against schema name.
+
+* ``table`` (optional): regex to match against table name.
+
+* ``privileges`` (required): zero or more of ``SELECT``, ``INSERT``,
```

Comment:
This is confusing. How does "zero" and "required" interact? Is this legal?
```
"privileges": ""
```
If so, then there should probably be a ``NONE`` privilege:
```
``privileges`` (required): either ``NONE`` or one or more of ...